### DISCUSSION TITLE
Event for logout from external provider

### DISCUSSION THREAD

**Discussion by joshdcu** [zitadel/zitadel#8798] [Q&A] [answered]
I want to know if a user failed to login to Zitadel using an external provider. I looked at the [documentation for Zitadel's event API](https://zitadel.com/docs/guides/integrate/zitadel-apis/event-api) and thought it might be captured as an event. However, when I [fetched the types of events](https://zitadel.com/docs/apis/resources/admin/admin-service-list-event-types), I could only find an event for a _successful_ login...

```json
{
    "type": "user.human.externallogin.check.succeeded",
    "localized": {
        "key": "EventTypes.user.human.externallogin.check.succeeded",
        "localizedMessage": "External login succeeded"
    }
}
```
i.e, there is no corresponding `user.human.externallogin.check.failed`.

1. Is this event named differently? I couldn't find any other event related to failed external logins.
2. If this event doesn't exist, is there a reason? I can see we have a [comprehensive list of events for other kinds of login failures](https://zitadel.com/docs/guides/integrate/zitadel-apis/event-api#example-get-failed-login-attempt) and am curious why this is left out.

PS: I understand there is another way to get this information using Zitadel actions. We can use [the External Authentication flow and the Post Authentication trigger](https://zitadel.com/docs/apis/actions/external-authentication).

**Comment by hifabienne** [comment]
I quickly had a look at the code, we do not have the failed event.

@muhlemmer @livio-a can you tell why we don't have the failed event?

**Comment by muhlemmer** [comment, marked as answer]
Without looking at the code my best guess is that we don;t now who the user is if the authentication failed. Events in zitadel are created on an Aggregate. Each aggregate has a unique ID. The `user.human.externallogin.check.succeeded` event belongs to the `user` aggregate. The unique  ID (`aggregate_id`) is the user's ID.

So we can't push `user.human.externallogin.check.failed` event if we do not know the aggegate ID (User ID) to push the event to. We don't know the user ID because the external authentication did not complete successful.

**Reply by joshdcu** [reply]
Got it, that makes sense to me! Thanks for the explanation 😄
